### PR TITLE
Add snapshot Ceph snapshot class in k8s 1.13.3

### PR DESCRIPTION
**Pull request by rollandf**
- Add snapshot class yaml
- Wait for RBD plugin to be up before creating the snapshot class

### CODE REVIEW

**Comment by rollandf** [comment]
@igoihman please take a look

File: k8s/scripts/provision.sh [hunk 1]
```diff
@@ -171,10 +171,10 @@ kind: MasterConfiguration
 apiServerExtraArgs:
   runtime-config: admissionregistration.k8s.io/v1alpha1
   ${admission_flag}: Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota
-  feature-gates: "BlockVolume=true,CustomResourceSubresources=true,CSIBlockVolume=true"
+  feature-gates: "CustomResourceSubresources=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true"
```
Comment (igoihman) [inline k8s/scripts/provision.sh]: this is for 1.11 and below versions, this feature gate doesn't exist in these versions.

File: k8s/scripts/provision.sh [hunk 2]
```diff
@@ -152,12 +152,12 @@ kind: InitConfiguration
...
 apiServerExtraArgs:
   enable-admission-plugins: Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota
-  feature-gates: "BlockVolume=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true"
+  feature-gates: "CSIBlockVolume=true,VolumeSnapshotDataSource=true"
   allow-privileged: "true"
   runtime-config: admissionregistration.k8s.io/v1alpha1
 apiVersion: kubeadm.k8s.io/v1alpha3
 controllerManagerExtraArgs:
-  feature-gates: "BlockVolume=true,,CSIBlockVolume=true,VolumeSnapshotDataSource=true"
+  feature-gates: "CSIBlockVolume=true,VolumeSnapshotDataSource=true"
```
Comment (igoihman) [inline k8s/scripts/provision.sh]: is there a reason why BlockVolume was removed?

File: k8s/scripts/provision.sh [hunk 3]
```diff
@@ -171,10 +171,10 @@ kind: MasterConfiguration
 apiServerExtraArgs:
   runtime-config: admissionregistration.k8s.io/v1alpha1
   ${admission_flag}: Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota
-  feature-gates: "BlockVolume=true,CustomResourceSubresources=true,CSIBlockVolume=true"
+  feature-gates: "CustomResourceSubresources=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true"
   allow-privileged: "true"
 controllerManagerExtraArgs:
-  feature-gates: "BlockVolume=true,CSIBlockVolume=true"
+  feature-gates: "CSIBlockVolume=true,VolumeSnapshotDataSource=true"
```
Comment (igoihman) [inline k8s/scripts/provision.sh]: same

File: k8s/scripts/provision.sh (hunk 2 above)
Comment (rollandf) [reply k8s/scripts/provision.sh]: Reverted

File: k8s/scripts/provision.sh (hunk 3 above)
Comment (rollandf) [reply k8s/scripts/provision.sh]: Reverted

File: k8s/scripts/provision.sh (hunk 1 above)
Comment (rollandf) [reply k8s/scripts/provision.sh]: Reverted

**Comment by rollandf** [comment]
ping

File: k8s/scripts/ceph-csi.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-set -xe
+set -x
```
Comment (rmohr) [inline k8s/scripts/ceph-csi.sh]: Please leave `set -xe` here.

File: k8s/scripts/ceph-csi.sh [hunk 5]
```diff
@@ -15,6 +15,23 @@ kubectl --kubeconfig /etc/kubernetes/admin.conf create -f /tmp/ceph/csi-rbdplugi
 # Deploy RBD CSI driver
 kubectl --kubeconfig /etc/kubernetes/admin.conf create -f /tmp/ceph/csi-rbdplugin.yaml
 
-# Deploy Ceph secret and storageclass
+# Deploy Ceph secret, storageclass and snapshotclass
 kubectl --kubeconfig /etc/kubernetes/admin.conf create -f /tmp/ceph/ceph-secret.yaml
 kubectl --kubeconfig /etc/kubernetes/admin.conf create -f /tmp/ceph/ceph-storageclass.yaml
+
+sleep 10
+status=`kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods --all-namespaces --no-headers --field-selector status.phase!=Running --output=name | grep rbdplugin`
```
Comment (rmohr) [inline k8s/scripts/ceph-csi.sh]: You can do `set +e` here.

File: k8s/scripts/ceph-csi.sh [hunk 6]
```diff
@@ -15,6 +15,23 @@ kubectl --kubeconfig /etc/kubernetes/admin.conf create -f /tmp/ceph/csi-rbdplugi
...
+retry_counter=0
+while [[ $retry_counter -lt 30 && ! -z "$status" ]]; do
+    sleep 10
+    echo "Waiting for rbd plugin to be available..."
+    status=`kubectl --kubeconfig=/etc/kubernetes/admin.conf get pods --all-namespaces --no-headers --field-selector status.phase!=Running --output=name | grep rbdplugin`
+    retry_counter=$((retry_counter + 1))
+done
```
Comment (rmohr) [inline k8s/scripts/ceph-csi.sh]: and `set -e` after the while loop.

**Comment by rmohr** [comment]
@rollandf changes look good in general. Seems like there was an rpm update in the k8s rpm repositories which breaks the build:

```
You could try using --skip-broken to work around the problem
--
722 | Error: Package: kubelet-1.13.3-0.x86_64 (kubernetes)
723 | Requires: kubernetes-cni = 0.6.0
724 | Available: kubernetes-cni-0.3.0.1-0.07a8a2.x86_64 (kubernetes)
725 | kubernetes-cni = 0.3.0.1-0.07a8a2
726 | Available: kubernetes-cni-0.5.1-0.x86_64 (kubernetes)
727 | kubernetes-cni = 0.5.1-0
728 | Available: kubernetes-cni-0.5.1-1.x86_64 (kubernetes)
729 | kubernetes-cni = 0.5.1-1
730 | Available: kubernetes-cni-0.6.0-0.x86_64 (kubernetes)
731 | kubernetes-cni = 0.6.0-0
732 | Installing: kubernetes-cni-0.7.5-0.x86_64 (kubernetes)
733 | kubernetes-cni = 0.7.5-0
734 | You could try running: rpm -Va --nofiles --nodigest
```

**Comment by rmohr** [comment]
Seems to be related to this: https://github.com/kubernetes/kubernetes/issues/75683

**Comment by rmohr** [comment]
@rollandf I think if you set the kubernetes-cni version explicitly in https://github.com/kubevirt/kubevirtci/blob/master/k8s/scripts/provision.sh#L60 to `0.6.0` it should work again.

File: k8s/scripts/ceph-csi.sh (hunk 4 above)
Comment (rollandf) [reply k8s/scripts/ceph-csi.sh]: done

File: k8s/scripts/ceph-csi.sh (hunk 5 above)
Comment (rollandf) [reply k8s/scripts/ceph-csi.sh]: done

File: k8s/scripts/ceph-csi.sh (hunk 6 above)
Comment (rollandf) [reply k8s/scripts/ceph-csi.sh]: done

**Comment by rmohr** [comment]
/lgtm

@rollandf looks good. The push result shows this tag: `sha256:bc0f02d6b970650eb16d12f97e5aa1376b3a13b0ffed6227db98675be2ca1184`, could you update the README.md with this tag in another commit on the PR? Once it is there we can merge it.

**Comment by rmohr** [comment]
/lgtm

@igoihman any last words?

**Comment by rmohr** [comment]
@rollandf you can already try the image out wherever you need it.

**Comment by rollandf** [comment]
> @rollandf you can already try the image out wherever you need it.

Tested with the image, works fine

**Pull request merged**